### ISSUE TITLE
[quotemark] backticks break lookup types / use strict / octal escape sequences / enum members

### ISSUE THREAD

**Issue by ericbf** [palantir/tslint#4692]
### Bug Report

- __TSLint version__: 5.16.0
- __TypeScript version__: 3.4.2
- __Running TSLint via__: Atom

#### TypeScript code being linted

```ts
"use strict"

enum Hello {
    "world",
    "foo"
}

function hi() {
    "use strict"
}

type hello = typeof obj["key"]
type hello2 = InterfaceName["key"]

const octal = "\110\151"
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "quotemark": [true, "backtick"]
    }
}
```

#### Actual behavior

All the strings in the example are flagged, even though they must remain as single/double quotes in order to compile or otherwise behave correctly.

#### Expected behavior

None of the strings in the example are flagged.
